### PR TITLE
feat(clientLibs): Fix up client libs page

### PR DESCRIPTION
**Pull request by russorat**
Closes https://github.com/influxdata/influxdb/issues/15159

Describe your proposed changes here.
image

image

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/clientLibraries/components/ClientGoOverlay.tsx [hunk 1]
```diff
@@ -6,10 +6,40 @@ import ClientLibraryOverlay from 'src/clientLibraries/components/ClientLibraryOv
 
 // Constants
 import {clientGoLibrary} from 'src/clientLibraries/constants'
+import CodeSnippet from 'src/shared/components/CodeSnippet'
 
 const ClientGoOverlay: FunctionComponent<{}> = () => {
   const {name, url} = clientGoLibrary
+  const initializeClientCodeSnippet = `
```
Comment (drdelambre) [inline ui/src/clientLibraries/components/ClientGoOverlay.tsx]: These should be hoisted out of the render method (to line 10ish) for a small perf upgrade as they don't reference props

Comment (drdelambre) [reply ui/src/clientLibraries/components/ClientGoOverlay.tsx]: Also why aren't they defined in the constants file in the same way as in the C# example?

Comment (russorat) [reply ui/src/clientLibraries/components/ClientGoOverlay.tsx]: WIP

**Comment by drdelambre** [comment]
run `yarn test -u` to update the snapshots that are failing

File: ui/src/clientLibraries/components/ClientCSharpOverlay.tsx [hunk 2]
```diff
@@ -17,155 +29,48 @@ const ClientCSharpOverlay: FunctionComponent<{}> = () => {
         </a>
       </p>
       <br />
-      <h5>Installing package</h5>
+      <h5>Installing Package</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientCSharpOverlay.tsx]: Use imperatives in step-by-step processes.

~Installing Package~ -> Install Package

File: ui/src/clientLibraries/components/ClientCSharpOverlay.tsx [hunk 3]
```diff
@@ -17,155 +29,48 @@ const ClientCSharpOverlay: FunctionComponent<{}> = () => {
...
+        label="Code"
+      />
       <p>
         <b>Package Reference</b>
       </p>
-      <pre>
-        <code>&lt;PackageReference Include="InfluxDB.Client"/&gt;</code>
-      </pre>
+      <CodeSnippet copyText={packageReferenceCodeSnippet} label="Code" />
       <h5>Initializing the Client</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientCSharpOverlay.tsx]: ~Initializing the Client~ -> Initialize the Client

File: ui/src/clientLibraries/components/ClientCSharpOverlay.tsx [hunk 4]
```diff
@@ -17,155 +29,48 @@ const ClientCSharpOverlay: FunctionComponent<{}> = () => {
...
-          const string query = "from(bucket: \"my_bucket\") |> range(start:
-          -1h)";
-          <br />
-          <br />
-          var tables = await client.GetQueryApi().QueryAsync(query,
-          "someorgid");
-        </code>
-      </pre>
+      <CodeSnippet copyText={initializeClientCodeSnippet} label="C# Code" />
       <h5>Writing Data</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientCSharpOverlay.tsx]: ~Writing Data~ -> Write Data

File: ui/src/clientLibraries/components/ClientCSharpOverlay.tsx [hunk 5]
```diff
@@ -17,155 +29,48 @@ const ClientCSharpOverlay: FunctionComponent<{}> = () => {
...
       <p>
-        Data could be written using InfluxDB Line Protocol, Data Point or POCO
-      </p>
-      <p>
-        <b>InfluxDB Line Protocol</b>
+        <b>Option 1: Example for writing using InfluxDB Line Protocol</b>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientCSharpOverlay.tsx]: ~Option 1: Example for writing using InfluxDB Line Protocol~ -> Option 1: Use InfluxDB Line Protocol to write data

File: ui/src/clientLibraries/components/ClientCSharpOverlay.tsx [hunk 6]
```diff
@@ -17,155 +29,48 @@ const ClientCSharpOverlay: FunctionComponent<{}> = () => {
...
-          <br />
-        </code>
-      </pre>
+      <CodeSnippet
+        copyText={writingDataLineProtocolCodeSnippet}
+        label="C# Code"
+      />
       <p>
-        <b>Data Point</b>
+        <b>Option 2: Example for writing using a Data Point</b>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientCSharpOverlay.tsx]: ~Option 2: Example for writing using a Data Point~ -> Option 1: Use a Data Point to write data

File: ui/src/clientLibraries/components/ClientCSharpOverlay.tsx [hunk 7]
```diff
@@ -17,155 +29,48 @@ const ClientCSharpOverlay: FunctionComponent<{}> = () => {
...
-          <br />
-          &#125;
-          <br />
-        </code>
-      </pre>
+      <CodeSnippet copyText={writingDataDataPointCodeSnippet} label="C# Code" />
       <p>
-        <b>POCO</b>
+        <b>
+          Option 3: Example for writing using a POCO and corresponding Class
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientCSharpOverlay.tsx]: ~Option 3: Example for writing using a POCO and corresponding Class~ -> Option 3: Use POCO and corresponding Class to write data

File: ui/src/clientLibraries/components/ClientCSharpOverlay.tsx [hunk 8]
```diff
@@ -17,155 +29,48 @@ const ClientCSharpOverlay: FunctionComponent<{}> = () => {
...
-          <br />
-          &nbsp;&nbsp;[Column(IsTimestamp = true)] public DateTime Time &#123;
-          get; set; &#125;
-          <br />
-          &#125;
-        </code>
-      </pre>
+      <CodeSnippet copyText={writingDataPocoCodeSnippet} label="C# Code" />
+      <CodeSnippet copyText={pocoClassCodeSnippet} label="C# Code" />
+      <h5>Example for executing a Flux query</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientCSharpOverlay.tsx]: ~Example for executing a Flux query~ -> Execute a Flux query

File: ui/src/clientLibraries/components/ClientGoOverlay.tsx [hunk 9]
```diff
@@ -18,6 +23,11 @@ const ClientGoOverlay: FunctionComponent<{}> = () => {
           GitHub Repository
         </a>
       </p>
+      <br />
+      <h5>Initializing the Client</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientGoOverlay.tsx]: ~Initializing the Client~ -> Initialize the Client

File: ui/src/clientLibraries/components/ClientGoOverlay.tsx [hunk 10]
```diff
@@ -18,6 +23,11 @@ const ClientGoOverlay: FunctionComponent<{}> = () => {
...
+      <CodeSnippet copyText={initializeClientCodeSnippet} label="Go Code" />
+      <h5>Example Writing Data</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientGoOverlay.tsx]: ~Example Writing Data~ -> Write Data

File: ui/src/clientLibraries/components/ClientJSOverlay.tsx [hunk 11]
```diff
@@ -20,34 +27,11 @@ const ClientJSOverlay: FunctionComponent<{}> = () => {
       </p>
       <br />
       <h5>Initializing the Client</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJSOverlay.tsx]: ~Initializing the Client~ -> Initialize the Client

File: ui/src/clientLibraries/components/ClientJSOverlay.tsx [hunk 12]
```diff
@@ -20,34 +27,11 @@ const ClientJSOverlay: FunctionComponent<{}> = () => {
...
-          const &#123;promise, cancel&#125; =
-          client.queries.execute('someorgid', query)
-        </code>
-        <br />
-        <code>const csv = await promise</code>
-        <br />
-        <code>cancel() // Cancels request</code>
-      </pre>
+      <CodeSnippet copyText={initializeClientCodeSnippet} label="Code" />
       <h5>Writing Data</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJSOverlay.tsx]: ~Writing Data~ -> Write Data

File: ui/src/clientLibraries/components/ClientJSOverlay.tsx [hunk 13]
```diff
@@ -20,34 +27,11 @@ const ClientJSOverlay: FunctionComponent<{}> = () => {
...
-      <p>Data should be written using InfluxDB Line Protocol</p>
-      <pre>
-        <code>const data = '' // Line protocal string</code>
-        <code>
-          const response = await client.write.create('orgID', 'bucketID', data)
-        </code>
-      </pre>
+      <CodeSnippet copyText={writingDataLineProtocolCodeSnippet} label="Code" />
+      <h5>Example for executing a Flux query</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJSOverlay.tsx]: ~Example for executing a Flux query~ -> Execute a Flux query

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx [hunk 14]
```diff
@@ -17,159 +29,30 @@ const ClientJavaOverlay: FunctionComponent<{}> = () => {
         </a>
       </p>
       <br />
-      <h5>Adding dependency</h5>
+      <h5>Adding Dependency</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: ~Adding Dependency~ -> Add Dependency

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx [hunk 15]
```diff
@@ -17,159 +29,30 @@ const ClientJavaOverlay: FunctionComponent<{}> = () => {
...
-        <code>
-          dependencies &#123;
-          <br />
-          &nbsp;&nbsp;compile "com.influxdb:influxdb-client-java:1.1.0"
-          <br />
-          &#125;
-        </code>
-      </pre>
+      <CodeSnippet copyText={buildWithGradleCodeSnippet} label="Code" />
       <h5>Initializing the Client</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: ~Initializing the Client~ -> Initialize the Client

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx [hunk 16]
```diff
@@ -17,159 +29,30 @@ const ClientJavaOverlay: FunctionComponent<{}> = () => {
...
-        <code>
-          String query = "from(bucket: \"my_bucket\") |> range(start: -1h)";
-          <br />
-          <br />
-          List&lt;FluxTable&gt; tables = client.getQueryApi().query(query,
-          "someorgid");
-        </code>
-      </pre>
+      <CodeSnippet copyText={initializeClientCodeSnippet} label="Java Code" />
       <h5>Writing Data</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: ~Writing Data~ -> Write Data

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx [hunk 17]
```diff
@@ -17,159 +29,30 @@ const ClientJavaOverlay: FunctionComponent<{}> = () => {
...
-          <br />
-          <br />
-          &nbsp;&nbsp;@Column(timestamp = true)
-          <br />
-          &nbsp;&nbsp;Instant time;
-          <br />
-          &#125;
-        </code>
-      </pre>
+      <p>Option 1: Example for writing using InfluxDB Line Protocol</p>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: ~Option 1: Example for writing using InfluxDB Line Protocol~ -> Option 1: Use InfluxDB Line Protocol to write data

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx [hunk 18]
```diff
@@ -17,159 +29,30 @@ const ClientJavaOverlay: FunctionComponent<{}> = () => {
...
+      <CodeSnippet
+        copyText={writingDataLineProtocolCodeSnippet}
+        label="Java Code"
+      />
+      <p>Option 2: Example for writing using a Data Point</p>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: ~Option 2: Example for writing using a Data Point~ -> Option 1: Use a Data Point to write data

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx [hunk 19]
```diff
@@ -17,159 +29,30 @@ const ClientJavaOverlay: FunctionComponent<{}> = () => {
...
+      <CodeSnippet copyText={writingDataPointCodeSnippet} label="Java Code" />
+      <p>Option 3: Example for writing using a POJO and corresponding class</p>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: ~Option 3: Example for writing using a POJO and corresponding class~ -> Option 3: Use POJO and corresponding class to write data

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx [hunk 20]
```diff
@@ -17,159 +29,30 @@ const ClientJavaOverlay: FunctionComponent<{}> = () => {
...
+      <CodeSnippet copyText={writingDataPojoCodeSnippet} label="Java Code" />
+      <CodeSnippet copyText={pojoClassCodeSnippet} label="Java Code" />
+      <h5>Example for executing a Flux query</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: ~Example for executing a Flux query~ -> Execute a Flux query

File: ui/src/clientLibraries/components/ClientPythonOverlay.tsx [hunk 21]
```diff
@@ -20,72 +30,18 @@ const ClientPythonOverlay: FunctionComponent<{}> = () => {
       </p>
       <br />
       <h5>Installing Package</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientPythonOverlay.tsx]: ~Installing Package~ -> Install Package

File: ui/src/clientLibraries/components/ClientPythonOverlay.tsx [hunk 22]
```diff
@@ -20,72 +30,18 @@ const ClientPythonOverlay: FunctionComponent<{}> = () => {
...
-      <pre>
-        <code>pip install influxdb-client</code>
-      </pre>
+      <CodeSnippet copyText={initializePackageCodeSnippet} label="Code" />
       <h5>Initializing the Client</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientPythonOverlay.tsx]: ~Initializing the Client~ -> Initialize the Client

File: ui/src/clientLibraries/components/ClientPythonOverlay.tsx [hunk 23]
```diff
@@ -20,72 +30,18 @@ const ClientPythonOverlay: FunctionComponent<{}> = () => {
...
-      <pre>
-        <code>
-          query = 'from(bucket: "my_bucket") |> range(start: -1h)'
-          <br />
-          <br />
-          tables = client.query_api().query(query, org="someorgid")
-        </code>
-      </pre>
+      <CodeSnippet copyText={initializeClientCodeSnippet} label="Code" />
       <h5>Writing Data</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientPythonOverlay.tsx]: ~Writing Data~ -> Write Data

File: ui/src/clientLibraries/components/ClientPythonOverlay.tsx [hunk 24]
```diff
@@ -20,72 +30,18 @@ const ClientPythonOverlay: FunctionComponent<{}> = () => {
...
-          sequence = ["mem,host=host1 used_percent=23.43234543 1556896326",
-          <br />
-          &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"mem,host=host1
-          available_percent=15.856523 1556896326"]
-          <br />
-          <br />
-          write_client.write("bucketID", "orgID", sequence)
-        </code>
-      </pre>
+      <p>Option 1: Example for writing data using InfluxDB Line Protocol</p>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientPythonOverlay.tsx]: ~Option 1: Example for writing data using InfluxDB Line Protocol~ -> Option 1: Use InfluxDB Line Protocol to write data

File: ui/src/clientLibraries/components/ClientPythonOverlay.tsx [hunk 25]
```diff
@@ -20,72 +30,18 @@ const ClientPythonOverlay: FunctionComponent<{}> = () => {
...
+      <CodeSnippet copyText={writingDataLineProtocolCodeSnippet} label="Code" />
+      <p>Option 2: Example for writing data using a Data Point</p>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientPythonOverlay.tsx]: ~Option 2: Example for writing data using a Data Point~ -> Option 2: Use a Data Point to write data

File: ui/src/clientLibraries/components/ClientPythonOverlay.tsx [hunk 26]
```diff
@@ -20,72 +30,18 @@ const ClientPythonOverlay: FunctionComponent<{}> = () => {
...
+      <CodeSnippet copyText={writingDataPointCodeSnippet} label="Code" />
+      <p>Option 3: Example for writing data using a Batch Sequence</p>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientPythonOverlay.tsx]: ~Option 3: Example for writing data using a Batch Sequence~ -> Option 3: Use a Batch Sequence to write data

File: ui/src/clientLibraries/components/ClientPythonOverlay.tsx [hunk 27]
```diff
@@ -20,72 +30,18 @@ const ClientPythonOverlay: FunctionComponent<{}> = () => {
...
+      <CodeSnippet copyText={writingDataBatchCodeSnippet} label="Code" />
+      <h5>Example for executing a Flux query</h5>
```
Comment (sanderson) [inline ui/src/clientLibraries/components/ClientPythonOverlay.tsx]: ~Example for executing a Flux query~ -> Execute a Flux query

**Pull request merged**